nrpy/infrastructures/BHaH/CurviBoundaryConditions/CurviBoundaryConditions.py

```diff
@@ -86,7 +86,16 @@ def parity_conditions_symbolic_dot_products(
 Documented in: Tutorial-Start_to_Finish-Curvilinear_BCs.ipynb
 */
 """
-    return outstr + ccg.c_codegen(parity, lhs_strings)
+
+    # We enforce double or higher precision to avoid issues in finding
+    # the inner/outer BC indicies
+    fp_type = par.parval_from_str("fp_type")
+    type_alias = "double" if fp_type == "float" else "REAL"
+    calculation = ccg.c_codegen(
+        parity, lhs_strings, fp_type="double" if fp_type == "float" else fp_type
+    ).replace("REAL ", f"{type_alias} ")
```

Comment:
I have just adjusted `c_codegen()` to accept custom values for `fp_type_alias`, so please set `fp_type_alias=DOUBLE`, then declare `#define  DOUBLE double` at the top of C files and use it throughout the functions you need, and finally replace `REAL` with `DOUBLE` in the function interiors. This is a much cleaner and maintainable solution.

Reply:
replace is no longer required now that `fp_type_alias` is sufficiently generic and is no longer tied to an infrastructure.  this has been resolved and tested with the `nrpyelliptic` example where a supplemental definition of `double` is specified in the header.